### DISCUSSION TITLE
それ，mROS 2いごくんちゃうん？？

### DISCUSSION THREAD

**Discussion by takasehideki** [algyan/XIAOGYAN#55] [Ideas]
ていう決意表明:D
https://twitter.com/takasehideki/status/1656622519141687297

今月中には public になるんちゃう？しらんけど;(
https://github.com/mROS-base/mros2-esp32

**Comment by matsujirushi** [comment]
ESP32-C3に対応しているか否かが気になるところです。

**Reply by matsujirushi** [reply]
むぐぐ...
https://github.com/mROS-base/mros2-esp32#supported-environment

**Reply by takasehideki** [reply]
まだ持ってないし,,, てだけなのですが， `idf.py set-target esp32c3` からコケてしまっていてタヒんでいます,,, 
なにをやればサポートできるのだろう？？（esp-idf なんもわからん;((

**Reply by takasehideki** [reply]
`export IDF_TARGET=esp32c3` したら `idf.py build` までできた！！（うごいたとはいってない:D

**Reply by yamati-kz** [reply]
Seeed Studio XIAO ESP32C3で [mros2-esp32](https://github.com/mROS-base/mros2-esp32)  が動きました。
[Examples](https://github.com/mROS-base/mros2-esp32#examples)内の pub_float32 と[mros2-host-examples](https://github.com/mROS-base/mros2-host-examples) 内の mros2_sub_float32 との組み合わせ。

母艦 : Ubuntu 20.04.6 (JetPack 5.1.1), Nvidia Jetson Xavier NX Developer Kit
参考 : [mros2-esp32とM5Stackでサンプルプログラムを試す](https://zenn.dev/tasada038/articles/fdbf3e5ff7cd01)

公式の[build Examples](https://github.com/mROS-base/mros2-esp32#build-examples)に悩んだことが記載されていて
よく読まないとなと思いました。この部分です。
/* If an error occurs, try the following*/
  python -m pip install jinja2
/* --- */

**Comment by yamati-kz** [comment]
mros2_echoback_stringでのログを添付します。esp32c3を使用しています。
/
/
![mros2_echoback_string](https://github.com/algyan/XIAOGYAN/assets/135952435/8aa2e03b-1e4e-406d-bbde-8d29d3011775)

**Reply by takasehideki** [reply]
awesome thanks!!! 😍

**Reply by yamati-kz** [reply]
mros2_echoreply_stringのログを添付します。
母艦(ubuntu 20.04)からpublishしたものをesp32c3でsubscribe。そのまま母艦へpublishして折返して母艦でsubscribeできたログです。
ご参考になれば幸いです。
![echoreply_string](https://github.com/algyan/XIAOGYAN/assets/135952435/ada81964-b546-4342-a0df-270f026951c6)

**Reply by yamati-kz** [reply]
今回の検証には[【現地開催！プレゼント有】mROS 2 ハンズオン！【オンライン覗き見枠も有り】](https://algyan.connpass.com/event/259028/)を参考にさせていただいており、ありがとうございます。
